### PR TITLE
Update 60-sensor.hwdb

### PR DESCRIPTION
**Pull request by ChopperRob**
added support for the Lenovo IdeaPad D330-10IGM screen orientation

### CODE REVIEW

File: hwdb.d/60-sensor.hwdb [hunk 1]
```diff
@@ -501,6 +501,10 @@ sensor:modalias:acpi:SMO8500*:dmi:bvnLENOVO:*:pvrLenovoMIIX3-830:*
 sensor:modalias:acpi:BOSC0200*:dmi:*:svnLENOVO:pn81H3:*
  ACCEL_MOUNT_MATRIX=0, 1, 0; -1, 0, 0; 0, 0, 1
 
+# IdeaPad D330-10IGM
```
Comment (bluca) [inline hwdb.d/60-sensor.hwdb]: if this is just a variant of the D330, given the matrix is the same, then this should be merged in the above rule, eg:

```
# IdeaPad D330 and D330-10IGM
sensor:modalias:acpi:BOSC0200*:dmi:*:svnLENOVO:pn81H3:*
sensor:modalias:acpi:BOSC0200*:dmi:*:svnLENOVO:*:cvrLenovoideapadD330-10IGM:*
 ACCEL_MOUNT_MATRIX=0, 1, 0; -1, 0, 0; 0, 0, 1
```

**Comment by jwrdegoede** [comment]
> /cc @jwrdegoede for double checking

I agree that it would be best to merge the 2 entries.

**Comment by ChopperRob** [comment]
Thank you, i've updated the pull request.

**Review by bluca — APPROVED** [review]
lgtm

**Comment by bluca** [comment]
For your next PR, please rebase/squash the commit rather than adding a new one on top with fixups.

**Comment by howl** [comment]
I didn't saw this in time. I'm pretty sure this is due to the fact that exists model names 81H3 and 81MD for the same product version D330-10IGM.

81H3 is also 
Handle 0x0001, DMI type 1, 27 bytes
System Information
	Manufacturer: LENOVO
	Product Name: 81H3
	Version: Lenovo ideapad D330-10IGM
	Serial Number: ********
	UUID: ********-****-****-****-************
	Wake-up Type: Power Switch
	SKU Number: LENOVO_MT_81H3_BU_idea_FM_ideapad D330-10IGM
	Family: ideapad D330-10IGM

Now a days even exists D330-10IGL that looks very similar but is only available in Japan.

All theese word to only say that we could then get rid "sensor:modalias:acpi:BOSC0200*:dmi:*:svnLENOVO:pn81H3:*" leaving only product vendor check for both 81H3 and 81MD. Should be interesting if this should be also used for framebuffer quirks and have all devices covered.

**Comment by howl** [comment]
@ChopperRob could you check if you need also this https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/gpu/drm/drm_panel_orientation_quirks.c ?

Panel in 81H3 is a horizontal one, if you use other DE nowadays than GNOME the desktop will be not orientated right to the keyboard by default. GNOME looks right just because it reorientates the screen with the accel data. If that is the same situation should be needed for your device to get rid 		  DMI_EXACT_MATCH(DMI_PRODUCT_NAME, "81H3"), from the actual 210 line.

Also you could notice that the initramfs console output is bad orientated, that could also be corrected.

@jwrdegoede has a lot of experience in this :D

**Comment by ChopperRob** [comment]
@howl i'm not exactly sure how i should check if i need it.
But the orientation is now only fixed after logging into GNOME. So the GNOME logon screen and the boot screen/console before GNOME is still incorrect. (is that the initramfs console?)

**Comment by jwrdegoede** [comment]
@ChopperRob:

> i'm not exactly sure how i should check if i need it. But the orientation is now only fixed after logging into GNOME. So the GNOME logon screen and the boot screen/console before GNOME is still incorrect. (is that the initramfs console?)

Ok, so this sounds like your variant of the D330 has a native portrait-mode screen (rather then the standard landscape-mode screen which normal laptops have), which means that when used in laptop mode (which has the screen in landscape-mode) everything, including the text messages during boot, is rotated 90 degrees and we need to fix this in software.

So to fix (workaround) this permanently for you and other users of the same model we will need to add an entry for your device to: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/drivers/gpu/drm/drm_panel_orientation_quirks.c as @howl already suggested.

But first lets test this does the right thing, by setting the same quirk temporarily from the kernel commandline.

First of all, please run: `ls /sys/class/drm`, I'm going to assume that you will see a `card0-DSI-1` in the output there.

Assuming I got that right then adding `video=DSI-1:panel_orientation=right_side_up` to the kernel commandline should fix the initramfs text, please see your distro's documentation for how to add extra arguments to the kernel commandline. You can verify of adding this to the cmdline worked by doing `cat /proc/cmdline` after rebooting.
Assuming the addition to the kernel commandline worked, then your text messages should be oriented the right way now.

If you don't have a `card0-DSI-1` entry in the output of `ls /sys/class/drm` then you probably will have a `card0-eDP-1` entry there, in that case you should try adding `video=eDP-1:panel_orientation=right_side_up` to the kernel commandline instead.

If things work on the kernel-cmdline then I'm going to need some more info to prepare a patch for the kernel to make this a permanent quirk for your model, please provide the output of the following commands:
```
cat /sys/class/drm/card0-DSI-1/modes
grep . /sys/class/dmi/id/* 2> /dev/null
```

**Comment by ChopperRob** [comment]
Hi @jwrdegoede, i indeed have the card0-DSI-1.
I've added the parameters to the kernel commandline and this fixed the orientation during boot.

Here are the outputs of the requested commands.

cat /sys/class/drm/card0-DSI-1/modes
```
1200x1920
```

grep . /sys/class/dmi/id/* 2> /dev/null
```
/sys/class/dmi/id/bios_date:10/30/2019
/sys/class/dmi/id/bios_release:1.37
/sys/class/dmi/id/bios_vendor:LENOVO
/sys/class/dmi/id/bios_version:8NCN37WW
/sys/class/dmi/id/board_asset_tag:NO Asset Tag
/sys/class/dmi/id/board_name:LNVNB161216
/sys/class/dmi/id/board_vendor:LENOVO
/sys/class/dmi/id/board_version:SDK0J91189 WIN
/sys/class/dmi/id/chassis_asset_tag:NO Asset Tag
/sys/class/dmi/id/chassis_type:32
/sys/class/dmi/id/chassis_vendor:LENOVO
/sys/class/dmi/id/chassis_version:Lenovo ideapad D330-10IGM
/sys/class/dmi/id/ec_firmware_release:1.37
/sys/class/dmi/id/modalias:dmi:bvnLENOVO:bvr8NCN37WW:bd10/30/2019:br1.37:efr1.37:svnLENOVO:pn81MD:pvrLenovoideapadD330-10IGM:rvnLENOVO:rnLNVNB161216:rvrSDK0J91189WIN:cvnLENOVO:ct32:cvrLenovoideapadD330-10IGM:
/sys/class/dmi/id/product_family:ideapad D330-10IGM
/sys/class/dmi/id/product_name:81MD
/sys/class/dmi/id/product_sku:LENOVO_MT_81MD_BU_idea_FM_ideapad D330-10IGM
/sys/class/dmi/id/product_version:Lenovo ideapad D330-10IGM
/sys/class/dmi/id/sys_vendor:LENOVO
/sys/class/dmi/id/uevent:MODALIAS=dmi:bvnLENOVO:bvr8NCN37WW:bd10/30/2019:br1.37:efr1.37:svnLENOVO:pn81MD:pvrLenovoideapadD330-10IGM:rvnLENOVO:rnLNVNB161216:rvrSDK0J91189WIN:cvnLENOVO:ct32:cvrLenovoideapadD330-10IGM:
```

**Comment by howl** [comment]
Offtopic: There are more recent BIOS solving some security issues, flash them at your own risk.

It's a 81MD as guessed, same product and chassis version, perhaps I will submit a pr changing your cvr to pvr to match how most of other devices match the string. The changes are the same needed for the 81H3 so the pr merging the matches for D330 in only one line are also right and merged.

I think D330-10IGL needs the same quirks but it has another product_version and only sold in Japan, so till someone reports something like @ChopperRob with the 81MD I think is better not to guess.

**Comment by howl** [comment]
@jwrdegoede forget to mention I think just renaming the comment to /* Lenovo Ideapad D330-IGM */and removing DMI_PRODUCT_NAME could be sufficient, but I prefer you take a look because that means this will be the first device there without product_name match in the dmi struct.

**Comment by ChopperRob** [comment]
> Offtopic: There are more recent BIOS solving some security issues, flash them at your own risk.
Thanks for the tip, i did notice it when posting the data.

**Comment by jwrdegoede** [comment]
@ChopperRob thank you for providing the info.

@howl
> I think just renaming the comment to /* Lenovo Ideapad D330-IGM */and removing DMI_PRODUCT_NAME could be sufficient

That is a good idea, the product-name match is there because either the 310 or 320 (don't remember) has versions both with and without a 90 degree rotated panel. Although I'm not sure if those 2 variants even have a different product-name and we also do a resolution check and 1280x800 != 800x1280 so the resolution check alone should also avoid applying the quirk to a non rotated panel. Anyways just some background info, as I said this is a good idea.

I have submitted a patch upstream dropping the matching on the DMI_PRODUCT_NAME:
https://lore.kernel.org/dri-devel/20210508111714.11697-1-hdegoede@redhat.com/T/#u

**Comment by howl** [comment]
@jwrdegoede I could see that @jnikula left me in the limbo and a patch to also set the rotation for hd models were omitted, without the product name would look something like this:

> 	...
> 	}, {	/* Lenovo Ideapad D330-10IGM (HD) */
> 		.matches = {
> 		  DMI_EXACT_MATCH(DMI_SYS_VENDOR, "LENOVO"),
> 		  DMI_EXACT_MATCH(DMI_PRODUCT_VERSION, "Lenovo ideapad D330-10IGM"),
> 		},
> 		.driver_data = (void *)&lcd800x1280_rightside_up,
> 	}, {	/* Lenovo Ideapad D330-10IGM (FHD) */
> 		.matches = {
> 		  DMI_EXACT_MATCH(DMI_SYS_VENDOR, "LENOVO"),
> 		  DMI_EXACT_MATCH(DMI_PRODUCT_VERSION, "Lenovo ideapad D330-10IGM"),
> 		},
> 		.driver_data = (void *)&lcd1200x1920_rightside_up,
> 	}, {
> 	...

**Comment by jwrdegoede** [comment]
@howl, thank you for letting me know about the missing entry for HD screens, I've created a new version of the patch which also adds the missing entry: https://patchwork.freedesktop.org/patch/432850/

**Pull request merged**